runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java

```diff
@@ -0,0 +1,221 @@
+		ItemID.STOLEN_GARNET_RING, ItemID.STOLEN_CIRCLET, ItemID.STOLEN_FAMILY_HEIRLOOM, ItemID.STOLEN_JEWELRY_BOX);
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private StealingArtefactsOverlay overlay;
+
+	@Getter(AccessLevel.PACKAGE)
+	private StealingArtefactState stealingArtefactState;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean inPortPiscariliusRegion;
+
+	private NPC captainKhaled;
+
+	@Override
+	protected void startUp()
+	{
+		overlayManager.add(overlay);
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invoke(() -> stealingArtefactState = StealingArtefactState.getState(client.getVar(Varbits.STEALING_ARTEFACT_STATE)));
+		}
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		client.clearHintArrow();
+		stealingArtefactState = null;
+		captainKhaled = null;
+		overlayManager.remove(overlay);
+	}
+
+	private boolean hasArtefact()
+	{
+		ItemContainer inventory = client.getItemContainer(InventoryID.INVENTORY);
+
+		if (inventory != null)
+		{
+			final Stream<Item> items = Arrays.stream(inventory.getItems());
+			return items.anyMatch(item -> ARTEFACT_ITEM_IDS.contains(item.getId()));
+		}
+		return false;
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		StealingArtefactState state = StealingArtefactState.getState(client.getVar(Varbits.STEALING_ARTEFACT_STATE));
+
+		if (state != StealingArtefactState.DELIVER_ARTEFACT)
+		{
+			stealingArtefactState = state;
+			if (state == StealingArtefactState.NO_TASK || state == StealingArtefactState.FAILURE)
+			{
+				client.clearHintArrow();
+			}
+			else
+			{
+				client.setHintArrow(stealingArtefactState.getWorldPoint());
+			}
+		}
+		else if (captainKhaled != null)
+		{
+			client.setHintArrow(captainKhaled);
+		}
+		else
+		{
+			client.clearHintArrow();
+		}
+	}
+
+	@Subscribe
+	public void onItemContainerChanged(final ItemContainerChanged event)
+	{
+		if (event.getItemContainer() == client.getItemContainer(InventoryID.INVENTORY))
+		{
+			if (hasArtefact())
+			{
+				stealingArtefactState = StealingArtefactState.DELIVER_ARTEFACT;
+			}
+			else if (stealingArtefactState == StealingArtefactState.DELIVER_ARTEFACT)
+			{
+				stealingArtefactState = StealingArtefactState.FAILURE;
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		if (event.getGameState() != GameState.LOGGED_IN)
+		{
+			captainKhaled = null;
+		}
+	}
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
+	{
+		if (event.getNpc().getId() == NpcID.CAPTAIN_KHALED_6972)
+		{
+			captainKhaled = event.getNpc();
+			if (stealingArtefactState == StealingArtefactState.DELIVER_ARTEFACT)
+			{
+				client.setHintArrow(captainKhaled);
+			}
+		}
+	}
+
+	@Subscribe
+	public void onNpcDespawned(NpcDespawned event)
+	{
+		if (event.getNpc().getId() == NpcID.CAPTAIN_KHALED_6972)
```

Comment:
This can just be a reference comparison with the stored npc.

Reply:
this isn't resolved

Reply:
ive been resolving convos as i go down in my own commit prior to push. should i switch to only doing that post push then?

Reply:
Resolve them once they've been resolved... That means pushed here so we can see that it's been resolved.